plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -224,20 +226,117 @@ public void testMigrateUnsupportedColumnType()
+
+        assertQuery("SELECT * FROM " + icebergTableName, "VALUES timestamp '" + timestamp + "'");
+
+        assertUpdate("DROP TABLE " + icebergTableName);
+    }
+
+    @Test
+    public void testMigrateTimestampMicrosTypeWithAvro()
+    {
+        String tableName = "test_migrate_timestamp_millis_avro" + randomNameSuffix();
+        String hiveTableName = "hive.tpch." + tableName;
+        Session session = timestampPrecisionSession(HiveTimestampPrecision.MICROSECONDS);
+
+        assertUpdate(session, "CREATE TABLE " + hiveTableName + " (x timestamp(6)) WITH (format='AVRO')");
+        assertUpdate(session, "INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 00:00:00.123456'", 1);
+        // Currently Hive maps timestamp(6) to timestamp(millis)
+        assertQuery("SELECT * FROM " + hiveTableName, "VALUES timestamp '2021-01-01 00:00:00.123'");
 
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '" + tableName + "')", "Migrating timestamp\\(3\\) type with AVRO format is not supported. Use \"timestamp\\(6\\)\" instead.");
+
+        assertUpdate("DROP TABLE " + hiveTableName);
+    }
+
+    @Test
+    public void testMigrateTimestampMillisTypeWithAvro()
+    {
+        String tableName = "test_migrate_timestamp_millis_avro" + randomNameSuffix();
+        String hiveTableName = "hive.tpch." + tableName;
+
+        assertUpdate("CREATE TABLE " + hiveTableName + " WITH (format='AVRO') AS SELECT timestamp '2021-01-01 00:00:00.000' x", 1);
         assertQuery("SELECT * FROM " + hiveTableName, "VALUES timestamp '2021-01-01 00:00:00.000'");
-        assertQueryFails("SELECT * FROM " + icebergTableName, "Not an Iceberg table: .*");
+
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '" + tableName + "')", "Migrating timestamp\\(3\\) type with AVRO format is not supported. Use \"timestamp\\(6\\)\" instead.");
 
         assertUpdate("DROP TABLE " + hiveTableName);
     }
 
+    @DataProvider
+    public static Object[][] timestampPrecisionsAndValuesParquet()
+    {
+        return new Object[][] {
+                //Hive strips TZ from timestamp, saves timestamp in UTC and additionally writer_tz to Parquet
+                //Iceberg reads only timestamp (which is in UTC)
+                //Hive has a setting hive.parquet.time-zone="America/Bahia_Banderas"
+                {HiveTimestampPrecision.MILLISECONDS, "2021-01-01 06:12:12.123", "2021-01-01 12:12:12.123"},
+                {HiveTimestampPrecision.MICROSECONDS, "2021-01-01 06:12:12.123456", "2021-01-01 12:12:12.123456"},
+                {HiveTimestampPrecision.NANOSECONDS, "2021-01-01 06:12:12.123456789", "2021-01-01 12:12:12.123457"} // Rounded as Parquet timestamp precision is 6
+        };
+    }
+
+    @Test(dataProvider = "timestampPrecisionsAndValuesParquet")
+    public void testMigrateColumnTypeTimestampOnFormatParquet(HiveTimestampPrecision precision, String input, String expected)
```

Comment:
1. similar method for orc is called `testMigrateTimestampTypeWithOrc`. call them more similarly, unless there indeed is difference that escaped my attention
2. why not have single test method that iterates over ORC, Parquet (and maybe Avro)?